perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -1463,35 +1463,19 @@ private void categorizeMembersForLightweightSynchronization(PerunSession sess, G
 				continue;
 			}
 
-			//try to find user from perun by login and extSource
+			//try to find user from perun by login and member extSource (need to use memberSource because loginSource is not saved by synchronization)
 			User user = null;
 			Candidate candidate = null;
 			try {
-				UserExtSource userExtSource = getPerunBl().getUsersManagerBl().getUserExtSourceByExtLogin(sess, loginSource, login);
+				UserExtSource userExtSource = getPerunBl().getUsersManagerBl().getUserExtSourceByExtLogin(sess, memberSource, login);
```

Comment:
Please, add comment to the code explaining why there must be memberSource and not loginSource. It's a bit tricky, so I rather have it stressed out in the code.